app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java

```diff
@@ -562,6 +562,24 @@ public void onItemSelected(final AdapterView<?> parent,
                 selectedSubtitleIndex = position;
                 break;
         }
+        onItemSelectedSetFileName();
+    }
+
+    private void onItemSelectedSetFileName() {
+        final String setSubtitleLanguageCode = subtitleStreamsAdapter.getItem(selectedSubtitleIndex)
+                .getLanguageTag();
+        final String fileName = FilenameUtils.createFilename(getContext(),
+                currentInfo.getName());
+        switch (dialogBinding.videoAudioGroup.getCheckedRadioButtonId()) {
+            case R.id.audio_button:
+            case R.id.video_button:
+                dialogBinding.fileName.setText(fileName);
+                break;
+            case R.id.subtitle_button:
+                dialogBinding.fileName.setText(fileName + getString(
+                        R.string.caption_file_name, fileName, setSubtitleLanguageCode));
```

Comment:
What is the `filename + `??
```suggestion
                dialogBinding.fileName.setText(getString(R.string.caption_file_name,
                        fileName, setSubtitleLanguageCode));
```